src/utils/args.ts

```diff
@@ -95,31 +102,28 @@ class ArgsNumber extends Args {
 
     }
 
-    static isValid = (value: ArgsNumber) => {
+    static isValid = (value: ArgsNumber): boolean => {
 
         // test if number
         // this shouldnt work for numbers where its string has "e" in it, but it does??
-        if (!/^\d*(\.\d*)?$/.test(value.value.toString()))
-            return;
+        if ( ! SIMPLE_NUM_REGEX.test(value.value.toString()) ) {
+            return false;
+        }
 
-        const decimals: number = value.decimals ?? {
-            NEAR: 24,
-            yocto: 0,
-            Tgas: 12,
-            gas: 0
-        }[value.unit]
+        // Try to initialize, otherwise assign an invalid value (-1)
+        let decimals: number = value.decimals ?? (value.unit ? unitToDecimals[value.unit] : -1);
```

Comment:
`decimals`' invalid state should be `undefined`, that's what what (for example) `convert` expects. If you strongly disagree with this, please adjust all functions dealing with `decimals` and also consider setting the invalid state to `0` instead of `-1`, so that we treat it as if it has no unit.